### PR TITLE
Added hint to caching conf file parameter

### LINKED ISSUE #11302: Keycloak is ignoring custom infinispan config
### Describe the bug

I want to run Keycloak 17 (Quarkus Edition) in HA mode with the provided infinispan. Because we are running Keycloak on serveral stages, I want to specify a infinispan cluster name. As I understood from the documentation I should configure this in the given infinispan config xml ./conf/cache-ispn.xml

I altered
    ` <transport lock-timeout="60000"/>`
to
`     <transport cluster="myClusterName" lock-timeout="60000"/>`

After that I ran .\kc.bat build --cache=ispn --cache-config-file=conf/cache-ispn.xml

and started up the server with .\kc.bat start

Sadly the output logging shows this:

```
[org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000078: Starting JGroups channel `ISPN`
[org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000094: Received new cluster view for channel ISPN: [MyHostName-14281|0] (1) [MyHostName-14281]
```
As seen in the logs, the cluster name still is the default "ISPN".

I already consulted the infinispan docs here: https://infinispan.org/docs/stable/titles/configuring/configuring.html

as well as the Keycloak docs:

https://www.keycloak.org/server/caching

https://www.keycloak.org/server/configuration

I also noticed that other configs like node names are ignored. 

Running .\kc.bat show-config shows that the infinispan config file is set correctly:

```
Current Mode: none
Runtime Configuration:
        kc.cache =  ispn (PropertiesConfigSource[source=file:/C:/tmp/keycloak-17.0.0/bin/../conf/keycloak.conf])
        kc.cache-config-file =  conf/cache-ispn.xml (PropertiesConfigSource[source=file:/C:/tmp/keycloak-17.0.0/bin/../conf/keycloak.conf])
        kc.config.args =  show-config (SysPropConfigSource)
        kc.db =  dev-file (PersistedConfigSource)
        kc.home.dir =  C:\tmp\keycloak-17.0.0\bin\.. (SysPropConfigSource)
        kc.hostname =  loclhost (PropertiesConfigSource[source=file:/C:/tmp/keycloak-17.0.0/bin/../conf/keycloak.conf])
        kc.http-enabled =  false (PropertiesConfigSource[source=jar:file:///C:/tmp/keycloak-17.0.0/lib/lib/main/org.keycloak.keycloak-quarkus-server-17.0.0.jar!/META-INF/keycloak.conf])
        kc.http-relative-path =  / (PersistedConfigSource)
        kc.metrics-enabled =  false (PersistedConfigSource)
        kc.quarkus-properties-enabled =  false (PersistedConfigSource)
        kc.show.config =  none (SysPropConfigSource)
        kc.version =  17.0.0 (SysPropConfigSource)
```

Am I missing something? Is the infinispan config loading broken in KC 17  Quarkus?

### Version

17.0.0

### Expected behavior

Infinispan config is loaded and custom cluster name is set corretly.

### Actual behavior

Infinispan config is ignored and the default cluster name (ISPN) is always set.

### How to Reproduce?

Try to set a custom cluster name in the infinispan config. 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kaiJwitt**
Added a hint to the --cache-config-file config option that you not need to specify the configuration folder

Closes #11302

### CODE REVIEW

**Comment by ahus1** [comment]
Thanks for contributing to this project with this pull request!

When creating PRs, this project asks in its [contributing guidelines](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md) to:

* create a GitHub issue first
* reference the GitHub issue in the commit message of the pull request with a message as the following:
  ```
  A brief descriptive summary

  Optionally, more details around how it was implemented

  Closes #1234
  ```
* update the description of this PR to include `Closes #1234`.

This will help collaboration on GitHub in two ways:

1. It populates the right-hand column on both the issue and the PR with a reference to the issue you're about to fix.
2. It will close the issue once the PR has been merged automatically.

All of this allows us to track all changes in the Git commit log to the issue on GitHub.

Please update your PR to align with this. Thanks!

**Comment by ahus1** [comment]
Thanks for aligning this PR! I now had a closer look: 

* the `conf/` in the new sentence should be placed in backticks like any other path in the file
* Looking at the code, [it (almost always?) prefixes the path of the conf folder,](https://github.com/keycloak/keycloak/blob/0908325e6d6ebbbbb9570d131f822641102035f7/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClusteringPropertyMappers.java#L37-L58) so the sentence should read something like this:
  ```
   The configuration file is relative to the `conf/` directory.
  ``` 
* Given this behavior, also the command line help should be updated as the current text ("Defines the file from which cache configuration should be loaded from.") doesn't reveal this logic. 

@DGuhr - do you agree with these findings?

**Comment by DGuhr** [comment]
@ahus1 Works for me. Just make sure to also edit the [Approvals for both scripts (win/*nix)](https://github.com/keycloak/keycloak/tree/main/quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help) we check against when changing the description [here](https://github.com/keycloak/keycloak/blob/main/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClusteringPropertyMappers.java#L37).

**Comment by ahus1** [comment]
@kaiJwitt - with the feedback from Dominik, I now ask you to:

* briefly verify that the configuration file is read relative to the `conf/` folder (so that I didn't miss something)
* update the docs accordingly (maybe you the suggested text above)
* update the command line help (like the previously suggested text, keep it short :-) )
* update the related test master files that Dominik mentioned to make the tests pass

When there are questions on the way, I'm happy to help.

**Comment by ahus1** [comment]
@kaiJwitt - the changes look good at first glance, I still need to se if the tests run successfully.

I see that there are now three commits in this PR. The contributing guidelines as for a single commit. Please squash your commits and force-push the changes. 

Thanks!

**Comment by ahus1** [comment]
@kaiJwitt - yes, unit tests will eventually run. As you're contributing to this repository for the first time, a maintainer will need to trigger the run. 

I'll have a look at the changes (probably tomorrow), and will then ask a maintainer to approve the GitHub action run with the unit tests.

**Comment by DGuhr** [comment]
@kaiJwitt Yes, they will as part of the "Quarkus Tests" step in the CI

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClusteringPropertyMappers.java
Comment (ahus1) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClusteringPropertyMappers.java:37]: There's a missing blank. Due to this, the new sentence doesn't start after a blank, but directly after the fullstop (".").

```suggestion
                        .description("Defines the file from which cache configuration should be loaded from. "
```

File: quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help/HelpCommandTest.testBuildHelp.unix.approved.txt
Comment (ahus1) [inline quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help/HelpCommandTest.testBuildHelp.unix.approved.txt:33]: The runtime will apply word-wrapping, therefore this test fails. 

The "The" will be at the previous line; and it will have a blank if it has been added to the Java code (as described above). Applies to all files, including the windows files. This is a manual edit, please run the tests locally (if possible), or ping me if I should run them.

```suggestion
                     Defines the file from which cache configuration should be loaded from. The
                       configuration file is relative to the 'conf/' directory.
```

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClusteringPropertyMappers.java
Comment (ahus1) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClusteringPropertyMappers.java:16]: Nitpick: while you're about to change the file anyway, please re-add the blank.

**Review by ahus1 — CHANGES_REQUESTED** [review]
I've reviewed the changes, unfortunately the unit tests fail. See below. I'm looking forward to an updated pull request when your time permits.

**Comment by kaiJwitt** [comment]
@ahus1 I've updated the pull request. On my local machine all tests are passing now. Can you re-run them?

**Review by ahus1 — APPROVED** [review]
I reviewed the changes and the look good to me. I also ran the updated unit tests on Windows and Linux and they passed. Thank you very much! I'll now ask a maintainer to approve the GitHub action run.

**Review by DGuhr — APPROVED** [review]

**Comment by kaiJwitt** [comment]
@ahus1 thank you so far for guiding me through this pull request. I've seen that the testsuits have been executed and sadly two of them were failing. 

I had a closer look at Base testsuite (wildfly, group1) and saw this:

`javax.ws.rs.ProcessingException: RESTEASY004655: Unable to invoke request: org.apache.http.conn.HttpHostConnectException: Connect to localhost:8543 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)`

This seems like an issue with the build server, right?

**Comment by ahus1** [comment]
@kaiJwitt - I had a look and these failures are unrelated to your changes. I'll see that a maintainer will merge this.

**Review by abstractj — APPROVED** [review]

**Pull request merged**
